### PR TITLE
codegen: close progrock writer

### PR DESCRIPTION
**Pull request by sipsma**
This seems to have been causing cloud progress to never finish running quite often.

cc @d3rp3tt3 @kpenfound

### CODE REVIEW

**Review by TomChv — APPROVED** [review]

File: cmd/codegen/main.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func ClientGen(cmd *cobra.Command, args []string) error {
 	var dialErr error
 	if propagateLogs {
 		progW, dialErr = progrock.DialRPC(ctx, "unix://"+nestedSock)
-		if err != nil {
+		if dialErr != nil {
 			return fmt.Errorf("error connecting to progrock: %w", err)
```
Comment (kpenfound) [inline cmd/codegen/main.go:51]:
```suggestion
			return fmt.Errorf("error connecting to progrock: %w", dialErr)
```
This one too, right?

Comment (gerhard) [reply cmd/codegen/main.go]: Good catch! There was another one gotcha down that function. LGTM now.

**Review by gerhard — APPROVED** [review]
OK to merge as soon as all checks go green.

**Comment by kpenfound** [comment]
Looks like it fixes it on my end!

File: cmd/codegen/main.go (hunk 1 above)
Comment (sipsma) [reply cmd/codegen/main.go]: Thank you!

**Pull request merged**